### PR TITLE
better trustline error handling; fix remove token

### PR DESCRIPTION
**Pull request by piyalbasu**

### CODE REVIEW

File: extension/src/popup/components/WarningMessages/index.tsx [hunk 1]
```diff
@@ -374,7 +374,12 @@ export const ScamAssetWarning = ({
     setIsSubmitting(false);
   };
 
-  return (
+  return isTrustlineErrorShowing ? (
+    createPortal(
+      <TrustlineError handleClose={() => closeOverlay()} />,
+      document.querySelector("#modal-root")!,
+    )
+  ) : (
```
Comment (piyalbasu) [inline extension/src/popup/components/WarningMessages/index.tsx:382]: the changes in this file replace the warning modal with a different error modal on tx submission fail

File: extension/src/popup/components/manageAssets/ChooseAsset/index.tsx [hunk 2]
```diff
@@ -96,6 +97,7 @@ export const ChooseAsset = ({ balances }: ChooseAssetProps) => {
                 getCanonicalFromAsset(code as string, issuer?.key as string)
               ],
             domain,
+            contract: contractId,
```
Comment (piyalbasu) [inline extension/src/popup/components/manageAssets/ChooseAsset/index.tsx:100]: this change makes sure we properly remove tokens

File: extension/src/popup/components/manageAssets/ManageAssetRowButton/index.tsx [hunk 3]
```diff
@@ -265,6 +289,13 @@ export const ManageAssetRowButton = ({
     }
   };
 
+  useEffect(() => {
+    if (submitStatus === ActionStatus.ERROR && isSigningWithHardwareWallet) {
+      setIsTrustlineErrorShowing(true);
+      setRowButtonShowing("");
+    }
+  }, [submitStatus, isSigningWithHardwareWallet]);
```
Comment (piyalbasu) [inline extension/src/popup/components/manageAssets/ManageAssetRowButton/index.tsx:297]: makes sure we're showing a error when hw wallet sign fails

File: extension/src/popup/components/manageAssets/ManageAssetRowButton/index.tsx [hunk 4]
```diff
@@ -137,7 +144,9 @@ export const ManageAssetRowButton = ({
         );
         trackChangeTrustline();
         dispatch(resetSubmission());
-        navigateTo(ROUTES.account);
+        if (successfulCallback) {
+          await successfulCallback();
```
Comment (piyalbasu) [inline extension/src/popup/components/manageAssets/ManageAssetRowButton/index.tsx:148]: makes sure we only nav away when there's no error with submission

**Review by aristidesstaffieri — APPROVED** [review]

**Pull request merged**